src/app/text-marking/text-marking.ts

```diff
@@ -99,10 +111,132 @@ export function markText(
+                // If the node was modified or inserted somewhere in the middle, apply animation
+                addItemAtIndex(markingCount, newTextContent)
+                newNode.style.animation = 'fillBackground 500ms linear forwards';
+            }else{
+                newNode.style.animation = 'fillBackground 500ms linear forwards';
+            }
+
+        }else{
+            newNode.style.animation = '';
         }
     }
+
+    markingCount++;
 }
 
+
+/**
+ * Monitors the progress of the `markText` function and triggers an action when the specified
+ * number of text markings is reached. This function is designed to be called after each invocation
+ * of the `markText` function to check if it has completed processing.
+ *
+ * @param {number} textMarkings - The current count of text markings processed by the `markText` function.
+ * @param {number} typoElements - The expected total count of text markings to be processed.
+ *
+ * When the number of processed text markings equals the total expected count (typoElements),
+ * this function triggers the `pushArrayItems` action and resets the marking count.
+ */
+function isLastCall(textMarkings: number, typoElements: number){
+    if(textMarkings === typoElements){
+        pushArrayItems()
+        markingCount = 0
+    }
+}
+
+/**
+ * Updates the state of the lastHighlightedMarkingWords array. This function is designed to be
+ * called when a specific condition is met, ensuring that the lastHighlightedMarkingWords array
+ * reflects the current state of the highlightedMarkingWords array.
+ *
+ * This action is typically triggered after the completion of a specific operation or task, ensuring
+ * that the arrays are synchronized for future comparisons.
+ */
+function pushArrayItems() {
+    lastHighlightedMarkingWords.length = 0
+    highlightedMarkingWords.forEach((item) => {
+        lastHighlightedMarkingWords.push(item);
+    });
+    highlightedMarkingWords.length = 0;
+}
+
+/**
+ * Removes an item from the lastHighlightedMarkingWords array at the specified index.
+ * @param {number} index - The index of the item to be removed from the array.
+ */
+function removeItemAtIndex(index: number){
+    if (index >= 0 && index < lastHighlightedMarkingWords.length) {
+
+      lastHighlightedMarkingWords.splice(index, 1);
+    } else {
+      console.error('Invalid index');
+    }
+}
+
+/**
+ * Adds a new item to the lastHighlightedMarkingWords array at the specified index.
+ * @param {number} index - The index at which to insert the new item.
+ * @param {string} newItem - The new item to be added to the array.
+ */
+function addItemAtIndex(index: number, newItem: string){
+    if (index >= 0 && index <= lastHighlightedMarkingWords.length) {
+
+      lastHighlightedMarkingWords.splice(index, 0, newItem);
+    } else {
+      console.error('Invalid index');
```

Comment:
Have you ever encountered a case in which this else case here is hit? The else cases in `removeItemAtIndex` and `addItemAtIndex` feel like they are trying to cover for some silent bugs that might be in `makeTextMarkings`.
That is, why would the indices be out of bounds in these two methods? The answer to this can perhaps be integrated in the `makeTextMarkings` so we can have less uncertainty regarding indices and so on.